src/Elements/types.jl

```diff
@@ -0,0 +1,180 @@
+const HEADERS   = ["#", "##", "###", "####", "#####", "######"]
+const EMPHASIS  = ["*", "**"]
+
+const CATEGORIES = [:module,    :function, :method, :macro, :type,
+                    :typealias, :global,   :symbol, :tuple, :aside]
+
+
+## Config Nodes
+abstract ConfigNode
+
+type Config <: ConfigNode
+    objfilter               :: Any
+    style_obj               :: ASCIIString
+
+    const defaults = Dict{Symbol, Any}(
+        :objfilter                  => CATEGORIES,
+        :style_obj                  => "####",
```

Comment:
> section within page: this? Is that possible?

I answered that in the above note.

> Note we should be able to pass arbitrary keyword arguments to a node to add config data. Avoids needing to explicitly write config(...) every time.

I had that in the previous version with the `title(), objs()` stuff.  Anyway, I think that can be done but what I do not like is that kwargs come at the end - looks not so clear with inheritat configuration.

```julia
section("Docile-API",
    page("Cache", config(:mkdocs_yml=true, style_obj="###"),
        "# Cache",
        "## Macros",
        "Some intermediate text describing more about macros.",
        (Cache, config(objfilter=[:macro])),
        "## Methods",
        (Cache, config(objfilter=[:method])),
        ),
    page("Cache",
        "# Cache",
        "## Macros",
        "Some intermediate text describing more about macros.",
        (Cache, config(objfilter=[:macro])),     # This inheriates the style_obj="### from page at the end
        "## Methods",
        (Cache, config(objfilter=[:method]));
        :mkdocs_yml=true, style_obj="###"      # HERE 
        ),
    ),

```

Or do you know how to allow arbitrary keyword arguments to a node kile this:

page("PageName", a=15, "# Header", anotherkwarg="IsThisPossible", Cache)